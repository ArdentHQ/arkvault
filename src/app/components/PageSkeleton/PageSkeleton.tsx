import React from "react";
import { Page, Section } from "@/app/components/Layout";
import { Skeleton } from "@/app/components/Skeleton";

export const PageSkeleton: React.FC = () => (
	<div data-testid="PageSkeleton">
		<Page navbarVariant="logo-only" title="Payvo Wallet">
			<Section className="-mt-5 flex flex-1 flex-col text-center md:mt-0 md:justify-center">
				<div className="mx-auto hidden w-96 md:block">
					<Skeleton height={230} />
				</div>
				<div className="flex flex-col items-center md:mt-8">
					<h2>
						<Skeleton width={100} height={32} />
					</h2>
					<p>
						<Skeleton width={240} height={24} />
					</p>
					<div className="mt-6 w-full sm:w-40 md:mt-8">
						<Skeleton height={160} />
					</div>
					<p className="mt-8 md:mt-16">
						<Skeleton width={280} height={24} />
					</p>
				</div>
			</Section>
		</Page>
	</div>
);
