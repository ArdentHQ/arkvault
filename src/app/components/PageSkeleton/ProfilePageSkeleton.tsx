import React from "react";
import { Skeleton } from "@/app/components/Skeleton";
import { Section } from "@/app/components/Layout";
import { Logo } from "@/app/components/Logo";

export const ProfilePageSkeleton: React.FC = () => (
	<div className="relative flex min-h-screen flex-col" data-testid="ProfilePageSkeleton">
		<div className="sticky inset-x-0 top-0 border-b border-theme-secondary-300 dark:border-theme-secondary-800">
			<div className="relative flex h-21">
				<div className="flex flex-1 justify-between px-8 sm:ml-12">
					<div className="my-auto mr-4 flex h-11 w-11 items-center justify-center rounded-xl bg-theme-success-600 pl-0.5 text-white">
						<Logo height={28} />
					</div>

					<div className="flex">
						<div className="my-auto flex items-center space-x-4">
							<Skeleton height={40} width={40} />
							<div className="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800" />
						</div>

						<div className="my-auto ml-4 hidden items-center space-x-4 sm:flex">
							<Skeleton height={40} width={40} />
							<div className="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800" />
							<Skeleton height={40} width={40} />
							<div className="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800" />
						</div>

						<div className="my-auto ml-4 flex items-center">
							<div className="mx-4 hidden space-y-2 md:block">
								<p>
									<Skeleton width={80} height={16} />
								</p>
								<p>
									<Skeleton width={80} height={16} />
								</p>
							</div>

							<Skeleton circle width={44} height={44} />
						</div>
					</div>
				</div>
			</div>
		</div>

		<div className="flex flex-1 flex-col">
			<Section>
				<div className="w-full">
					<Skeleton height={24} />
				</div>
				<div className="my-4 w-2/3">
					<Skeleton height={24} />
				</div>
				<div className="w-1/4">
					<Skeleton height={24} />
				</div>
			</Section>
		</div>
	</div>
);
