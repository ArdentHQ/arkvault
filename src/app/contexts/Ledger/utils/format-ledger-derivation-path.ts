import { LedgerDerivationScheme } from "@/app/contexts/Ledger/Ledger.contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
