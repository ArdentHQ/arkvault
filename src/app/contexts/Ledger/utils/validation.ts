import semver from "semver";
import { Coins } from "@payvo/sdk";
import { minVersionList } from "@/app/contexts/Ledger/Ledger.contracts";

export const hasRequiredAppVersion = async (coin: Coins.Coin) => {
	const coinId = coin.network().coin();

	if (minVersionList[coinId]) {
		const currentVersion = await coin.ledger().getVersion();

		if (semver.lt(currentVersion, minVersionList[coinId])) {
			return false;
		}
	}

	return true;
};
