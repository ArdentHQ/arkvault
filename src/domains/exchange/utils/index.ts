export * from "./assertions";
export * from "./errors";
