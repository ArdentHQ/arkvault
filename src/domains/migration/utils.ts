import { MigrationTransactionStatus } from "@/domains/migration/migration.contracts";

export const getIcon = (status: MigrationTransactionStatus) => {
	if (status === MigrationTransactionStatus.Confirmed) {
		return {
			color: "text-theme-success-600",
			name: "CircleCheckMark",
		};
	}

	return {
		color: "text-theme-warning-300",
		name: "Clock",
	};
};
