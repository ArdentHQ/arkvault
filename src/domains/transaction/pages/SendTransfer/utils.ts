export const getFeeType = (numberOfRecipients: number): "multiPayment" | "transfer" =>
	numberOfRecipients > 1 ? "multiPayment" : "transfer";
