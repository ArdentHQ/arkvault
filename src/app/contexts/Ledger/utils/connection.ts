import { Coins } from "@ardenthq/sdk";
import retry, { AbortError, Options } from "p-retry";

import { formatLedgerDerivationPath } from "./format-ledger-derivation-path";
import { hasRequiredAppVersion } from "./validation";

export const accessLedgerDevice = async (coin: Coins.Coin) => {
	try {
		await coin.__construct();
		await coin.ledger().connect();
	} catch (error) {
		// If the device is open, continue normally.
		// Can be triggered when the user retries ledger connection.
		if (error.message !== "The device is already open.") {
			throw error;
		}
	}
};

export const accessLedgerApp = async ({ coin }: { coin: Coins.Coin }) => {
	await accessLedgerDevice(coin);

	if (!(await hasRequiredAppVersion(coin))) {
		throw new Error("VERSION_ERROR");
	}

	// Ensure that the app is accessible.
	await coin.ledger().getPublicKey(
		formatLedgerDerivationPath({
			coinType: coin.config().get<number>("network.constants.slip44"),
		}),
	);
};

export const persistLedgerConnection = async ({
	coin,
	options,
	hasRequestedAbort,
}: {
	coin: Coins.Coin;
	options: Options;
	hasRequestedAbort: () => boolean;
}) => {
	const retryAccess: any = async (attempts: number) => {
		if (hasRequestedAbort() && attempts > 1) {
			throw new AbortError("CONNECTION_ERROR");
		}

		try {
			await accessLedgerApp({ coin });
		} catch (error) {
			// Abort on version error or continue retrying access.
			if (error.message === "VERSION_ERROR") {
				throw new AbortError("VERSION_ERROR");
			}

			throw error;
		}
	};

	await retry(retryAccess, options);
};
