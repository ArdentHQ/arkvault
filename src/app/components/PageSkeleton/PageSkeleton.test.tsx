import React from "react";

import { render } from "@/utils/testing-library";

import { PageSkeleton } from "./PageSkeleton";

describe("PageSkeleton", () => {
	it("should render without profile", () => {
		const { asFragment } = render(<PageSkeleton />);

		expect(asFragment()).toMatchSnapshot();
	});
});
