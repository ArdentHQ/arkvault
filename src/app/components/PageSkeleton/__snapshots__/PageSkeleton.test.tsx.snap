// Vitest Snapshot v1

exports[`PageSkeleton > should render without profile 1`] = `
<DocumentFragment>
  <div
    data-testid="PageSkeleton"
  >
    <div
      class="css-kmmje1"
    >
      <nav
        aria-labelledby="main menu"
        class="css-97i12m"
      >
        <div
          class="relative flex h-21"
        >
          <div
            class="flex flex-1 px-8 sm:ml-12"
          >
            <div
              class="my-auto flex items-center"
            >
              <button
                class="my-auto mr-4 flex h-11 w-11 cursor-pointer items-center justify-center rounded-xl bg-theme-primary-600 text-white outline-none focus:outline-none focus:ring-2 focus:ring-theme-primary-400"
                data-testid="NavigationBarLogo--button"
                type="button"
              >
                <span
                  class="relative"
                >
                  <svg
                    fill="none"
                    height="28"
                    viewBox="0 0 363 363"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      clip-path="url(#clip0_434_4442)"
                    >
                      <path
                        d="M174.182 352.325a279.256 279.256 0 0 1-25.627-20.536c-8.518-7.525-17.006-15.66-25.229-24.178a470.15 470.15 0 0 1-29.234-33.315 449.417 449.417 0 0 1-57.68-93.623 420.933 420.933 0 0 1-22.044-60.102A416.702 416.702 0 0 1 1.126 51.749c-.512-4.803-.303-10.73 5.9-13.842a393.077 393.077 0 0 1 54.42-19.94 459.697 459.697 0 0 1 52.646-12.195A374.409 374.409 0 0 1 176.15.333h11.6a374.406 374.406 0 0 1 62.057 5.44 459.723 459.723 0 0 1 52.646 12.195 393.328 393.328 0 0 1 54.422 19.939c6.203 3.115 6.41 9.034 5.898 13.842a416.672 416.672 0 0 1-13.242 68.822 421.133 421.133 0 0 1-22.044 60.102 449.395 449.395 0 0 1-57.68 93.623 470 470 0 0 1-29.234 33.315c-8.223 8.518-16.711 16.653-25.229 24.178a279.73 279.73 0 0 1-25.627 20.536 17.811 17.811 0 0 1-15.528 0h-.007zm6.769-230.087c-.412.547-28.363 29.571-48.772 50.764-7.035 7.318-13.229 13.736-17.187 17.853a49.345 49.345 0 0 0-14.37 29.355 42.981 42.981 0 0 0 2.034 17.531 59.295 59.295 0 0 0 10.41 18.547 369.34 369.34 0 0 0 42.811 44.915 223.324 223.324 0 0 0 20.072 16.051 13.364 13.364 0 0 0 12.005 0 223.333 223.333 0 0 0 20.07-16.051 369.191 369.191 0 0 0 42.811-44.915 59.297 59.297 0 0 0 10.411-18.547 42.98 42.98 0 0 0 2.033-17.531 49.34 49.34 0 0 0-14.377-29.358l-14.811-15.385-1.779-1.848c-20.66-21.451-48.955-50.832-49.371-51.383a1.179 1.179 0 0 0-.995-.548 1.184 1.184 0 0 0-.995.548v.002zm1.408-69.483c.136.215.321.505.582.869.391.544 6.257 9.93 14.382 22.92 19.488 31.162 52.102 83.315 60.679 96.014 6.342 9.377 14.353 14.77 21.974 14.772 7.402 0 18.108-5.031 27.079-29.067a332.382 332.382 0 0 0 19.124-81.095c.409-3.84.243-8.568-4.713-11.056a314.32 314.32 0 0 0-43.468-15.926 367.237 367.237 0 0 0-42.052-9.743 299.13 299.13 0 0 0-49.568-4.345h-8.871a299.13 299.13 0 0 0-49.568 4.345 367.225 367.225 0 0 0-42.051 9.743 314.427 314.427 0 0 0-43.469 15.926c-4.951 2.486-5.12 7.213-4.712 11.056a332.313 332.313 0 0 0 19.124 81.095c4.952 13.265 13.67 29.067 27.078 29.067 7.624 0 15.63-5.384 21.974-14.772 8.57-12.687 41.169-64.811 60.643-95.955l.032-.053c8.124-12.993 13.995-22.38 14.384-22.921.259-.364.455-.657.58-.87.209-.327.311-.491.416-.491.105 0 .214.16.421.487zm-55.455 149.533l19.652-20.614h69.159l19.649 20.614h-108.46zm34.197-35.377l20.162-20.871 20.159 20.871h-40.321z"
                        fill="currentColor"
                      />
                    </g>
                  </svg>
                </span>
              </button>
              <span
                class="text-lg uppercase"
              >
                <strong>
                  ARK
                </strong>
                Vault
              </span>
            </div>
          </div>
        </div>
      </nav>
      <div
        class="flex flex-1 flex-col"
      >
        <div
          class="-mt-5 flex flex-1 flex-col text-center md:mt-0 md:justify-center css-ycib48"
        >
          <div
            class="mx-auto px-8 md:px-10 lg:container"
          >
            <div
              class="mx-auto hidden w-96 md:block"
            >
              <span>
                <span
                  class="flex max-w-full items-center overflow-hidden leading-none"
                >
                  <span
                    class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                    style="height: 230px;"
                  >
                    ‌
                  </span>
                  ‌
                </span>
              </span>
            </div>
            <div
              class="flex flex-col items-center md:mt-8"
            >
              <h2>
                <span>
                  <span
                    class="flex max-w-full items-center overflow-hidden leading-none"
                  >
                    <span
                      class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                      style="width: 100px; height: 32px;"
                    >
                      ‌
                    </span>
                    ‌
                  </span>
                </span>
              </h2>
              <p>
                <span>
                  <span
                    class="flex max-w-full items-center overflow-hidden leading-none"
                  >
                    <span
                      class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                      style="width: 240px; height: 24px;"
                    >
                      ‌
                    </span>
                    ‌
                  </span>
                </span>
              </p>
              <div
                class="mt-6 w-full sm:w-40 md:mt-8"
              >
                <span>
                  <span
                    class="flex max-w-full items-center overflow-hidden leading-none"
                  >
                    <span
                      class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                      style="height: 160px;"
                    >
                      ‌
                    </span>
                    ‌
                  </span>
                </span>
              </div>
              <p
                class="mt-8 md:mt-16"
              >
                <span>
                  <span
                    class="flex max-w-full items-center overflow-hidden leading-none"
                  >
                    <span
                      class="react-loading-skeleton css-1eg0sfo-skeletonStyles-Skeleton"
                      style="width: 280px; height: 24px;"
                    >
                      ‌
                    </span>
                    ‌
                  </span>
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
